clis/tcli/src/bin/tclid/coms.rs

```diff
@@ -0,0 +1,152 @@
+        // Only one connection at a time is needed, so no task spawning.
+        loop {
+            let mut command_buffer = String::new();
+            match wait_for_tx(&self.listener_chan.tx, listener.accept()).await {
+                Some((send_permit, Ok(stream))) => {
+                    let mut stream = BufReader::new(stream);
+                    stream.read_line(&mut command_buffer).await?;
+                    // Removing the trailing newline used as a message ending according to platform.
+                    #[cfg(unix)]
+                    command_buffer.pop();
+                    #[cfg(not(unix))]
+                    todo!("Windows support is not implemented yet");
+
+                    send_permit.send(IPCMessage::Command(command_buffer));
+
+                    if let Some(response) = self.listener_chan.rx.recv().await {
+                        match response {
+                            IPCMessage::Command(_) => {
+                                eprint!("The listener got a command from the event_loop which should never happen, so ignoring.");
+                            }
+                            IPCMessage::Response(response) => {
+                                stream.write_all(response.as_bytes()).await?
+                            }
+                            IPCMessage::QuitResponse(response) => {
+                                stream.write_all(response.as_bytes()).await?;
+                                break Ok(());
+                            }
+                        }
+                    }
+                }
+                _ => {
+                    eprintln!("Event loop had stopped responding to ipc requests, so stopping IPC listener");
+                    break Ok(());
+                }
+            }
+        }
+    }
+
+    /// Function for sending commands to the server. Utilized by the API so purposefully made non-async to avoid overhead.
+    /// Blocks while waiting for a response, but that's intended behavior, because the API does not use an async runtime.
+    /// Also retrieves the response from the IPC server.
+    /// Note that endlines are used to indicate the end of a message so the server knows when to stop reading and
+    /// send a response, otherwise the server would wait until the API closes the socket and won't be able to send a response.
+    /// This is also sort of a deadlock risk, so should be handled with care.
+    ///
+    /// # Arguments
+    ///
+    /// * `addr` - Address (more specifically path, but left address for clarity) of the IPC socket.
+    /// * `cmd` - Command to be sent to the daemon.
+    ///
+    /// # Returns
+    ///
+    /// A Result containing a response string.
+    pub fn send_command(addr: &Path, cmd: &str) -> Result<String> {
+        let mut response_buffer = String::new();
+        let mut stream = BlockingSocketStream::connect(addr)?;
+        writeln!(stream, "{}", cmd)?;
+        stream.read_to_string(&mut response_buffer)?;
```

Comment:
I am slightly confused about one thing. `stream.read_to_string(....)` is supposed to read all IPC contents until local Unix socket is closed (e.g. until EOF), `tclid` does seem to work, but I am unsure why this `read_to_string` call does not block until daemon is stopped/closes it's side of the socket.

Reply:
This was supposed to be blocking. I made the API blocking instead of using an async runtime for performance reasons. Anyway I've reworked everything so that TCLID doesn't use an async runtime at all which is actually solving a lot of issues.

Reply:
Oh I was not thinking about async runtime ;), I meant, that `read_to_string()` should not return until daemon closes the unix domain socket, which, to my understanding it never does. Therefore I was confused why the client does not hang forever on this line.